include/swift/AST/TypeCheckRequests.h

```diff
@@ -1744,6 +1744,47 @@ class ValueWitnessRequest
   void cacheResult(Witness value) const;
 };
 
+/// Computes initializer contexts for each of the parameters with default
+/// arguments within a parameter list.
+class DefaultArgumentInitContextsRequest
```

Comment:
Is it really worth making this a request? It looks like a request that's run for its side effects, which is something we would like to avoid having. Maybe it should at least be per-ParamDecl and not per-ParameterList?

Reply:
@slavapestov The reasoning for putting at the level of `ParameterList` is to allow us to access the parameter index when creating the contexts. We could make it per-ParamDecl, but then we'd need to iterate over the parameter list for each param to find its own index. Either that or start storing the index on ParamDecl. What do you think?

Reply:
@slavapestov On second thought, we can totally make this per-ParamDecl, we can just cache the other parameter's init contexts while we iterate. Does the updated code look good?